packages/host/app/lib/limited-set.ts

```diff
@@ -0,0 +1,64 @@
+/* A bounded set that automatically removes the oldest entries when it reaches a maximum size */
+export default class LimitedSet<T> implements Iterable<T> {
```

Comment:
would it not be easier to just proxy `Set` and add the new capability in  methods that need it? what way you don't have to reimplement the methods that don't change.